From 1f92f6019f8befed6da1d11ccfb71494083b8ddb Mon Sep 17 00:00:00 2001
From: Ben Peled <bpeled@marvell.com>
Date: Mon, 15 Apr 2019 10:45:41 +0300
Subject: [PATCH 161/386] arm64: dts: marvell: Add cpu clock node on Armada
 7K/8K

Add cpu clock node on AP806

Change-Id: Ibd88a0e2884078d77b08471286e702b033161748
Signed-off-by: Gregory CLEMENT <gregory.clement@bootlin.com>
Signed-off-by: Ben Peled <bpeled@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/7216
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Nadav Haklai <Nadav.Haklai@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi |  4 ++++
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi      | 14 ++++++++++++++
 arch/arm64/boot/dts/marvell/armada-ap80x.dtsi      |  2 +-
 3 files changed, 19 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi
index 27a6994a3aeb..34ecc8b07703 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi
@@ -20,6 +20,7 @@
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x000>;
 			enable-method = "psci";
+			clocks = <&cpu_clk 0>;
 			cpu-idle-states = <&CPU_SLEEP_0>;
 			i-cache-size = <0xc000>;
 			i-cache-line-size = <64>;
@@ -34,6 +35,7 @@
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x001>;
 			enable-method = "psci";
+			clocks = <&cpu_clk 0>;
 			cpu-idle-states = <&CPU_SLEEP_0>;
 			i-cache-size = <0xc000>;
 			i-cache-line-size = <64>;
@@ -48,6 +50,7 @@
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x100>;
 			enable-method = "psci";
+			clocks = <&cpu_clk 1>;
 			cpu-idle-states = <&CPU_SLEEP_0>;
 			i-cache-size = <0xc000>;
 			i-cache-line-size = <64>;
@@ -62,6 +65,7 @@
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x101>;
 			enable-method = "psci";
+			clocks = <&cpu_clk 1>;
 			cpu-idle-states = <&CPU_SLEEP_0>;
 			i-cache-size = <0xc000>;
 			i-cache-line-size = <64>;
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 52c9e919afea..175c64a73484 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -12,4 +12,18 @@
 
 / {
 	model = "Marvell Armada AP806";
+	AP_NAME {
+		config-space@f0000000 {
+			ap_syscon1:system-controller@6f8000 {
+				compatible = "syscon", "simple-mfd";
+				reg = <0x6f8000 0x1000>;
+
+				cpu_clk: clock-cpu {
+					compatible = "marvell,ap806-cpu-clock";
+					clocks = <&ap_clk 0>, <&ap_clk 1>;
+					#clock-cells = <1>;
+				};
+			};
+		};
+	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi b/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
index 45961be442ac..027317a3dbac 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
@@ -400,7 +400,7 @@
 				status = "disabled";
 			};
 
-			ap_syscon: system-controller@6f4000 {
+			ap_syscon0: system-controller@6f4000 {
 				compatible = "syscon", "simple-mfd";
 				reg = <0x6f4000 0x2000>;
 
-- 
2.17.1

